lib/screenplay/suppressed_predictions/suppressed_prediction.ex

```diff
@@ -0,0 +1,172 @@
+  @jfk_umass_braintree_place "jfk_umass_braintree_platform"
+  @green_line_routes ["Green-B", "Green-C", "Green-D", "Green-E"]
+  @valid_silver_line_routes ["741", "742", "743", "746"]
+
+  @derive {Jason.Encoder, except: [:__meta__]}
+
+  @type t() :: %__MODULE__{
+          location_id: String.t(),
+          route_id: String.t(),
+          direction_id: integer(),
+          clear_at_end_of_day: boolean()
+        }
+
+  @primary_key false
+  schema "suppressed_predictions" do
+    field(:location_id, :string, primary_key: true)
+    field(:route_id, :string, primary_key: true)
+    field(:direction_id, :integer, primary_key: true)
+    field(:clear_at_end_of_day, :boolean)
+
+    timestamps(type: :utc_datetime)
+  end
+
+  def changeset(suppressed_prediction, attrs \\ %{}) do
+    places = Places.get()
+
+    suppressed_prediction
+    |> cast(attrs, [
+      :location_id,
+      :route_id,
+      :direction_id,
+      :clear_at_end_of_day
+    ])
+    |> validate_required([
+      :location_id,
+      :route_id,
+      :direction_id,
+      :clear_at_end_of_day
+    ])
+    |> validate_number(:direction_id,
+      less_than_or_equal_to: 1,
+      greater_than_or_equal_to: 0,
+      message: "Location Direction ID must be 0 or 1"
+    )
+    |> validate_edge_cases()
+    |> validate_location(places)
+    |> validate_routes_with_location_id(places)
+  end
+
+  defp validate_edge_cases(changeset) do
+    location_id = get_field(changeset, :location_id)
+    route_id = get_field(changeset, :route_id)
+
+    cond do
+      # JFK/UMass has the Ashmont/Braintree split, we ask for those stop ids rather than the parent id
+      location_id == "place-jfk" ->
+        add_error(
+          changeset,
+          :location_id,
+          "Please provide either the Ashmont or Braintree platform location_id: (#{@jfk_umass_ashmont_place} or #{@jfk_umass_braintree_place})"
+        )
+
+      # We don't handle separate Green Line and Silver Line branching route_ids
+      # That is populated when we pass data to Transit Data
+      # For internal use we just use "Green" or "Silver" to show all green/silver line routes
+      route_id in @green_line_routes ->
+        add_error(
+          changeset,
+          :route_id,
+          "Please provide just `Green` as the route_id for handling all Green Line routes"
+        )
+
+      true ->
+        changeset
+    end
+  end
+
+  defp validate_location(changeset, places) do
+    location_id = get_field(changeset, :location_id)
+
+    if places |> Enum.any?(&(&1.id == location_id)) do
+      changeset
+    else
+      add_error(changeset, :location_id, "Location `#{location_id}` does not exist")
+    end
+  end
+
+  defp validate_routes_with_location_id(changeset, places) do
+    location_id = get_field(changeset, :location_id)
+
+    if location_id in [@jfk_umass_ashmont_place, @jfk_umass_braintree_place] do
+      "place-jfk"
+    else
+      location_id
+    end
+    |> validate_routes(changeset, places)
```

Comment:
nit: Since this function is small and single-purpose, we could just fold it into `validate_routes` by assigning a variable to the result of the `if`, if that seems clearer.

Reply:
Moved it into the validate_routes, makes sense. 

I initially just wanted the function flow to be super explicit about it passing in a potentially transformed location_id but it might just be a tad overkill.